File: Front/src/types/produto.ts
export interface Produto {
  id: string;
  sku: string;
  nome: string;
  descricao: string | null;
  categoriaId: string;
  unidadeMedida: string;
  codigoBarras: string | null;
  estoqueMinimo: number;
}

export interface CriarProdutoRequest {
  sku: string;
  nome: string;
  descricao: string | null;
  categoriaId: string;
  unidadeMedida: string;
  codigoBarras: string | null;
  estoqueMinimo: number;
}

/** SKU não é editável — é o identificador estável do produto (RN03). */
export interface AtualizarProdutoRequest {
  nome: string;
  descricao: string | null;
  categoriaId: string;
  unidadeMedida: string;
  codigoBarras: string | null;
  estoqueMinimo: number;
}

File: GestaoArmazem/src/GestaoArmazem.Domain/Entities/Produto.cs
namespace GestaoArmazem.Domain.Entities;

public class Produto
{
    public Guid Id { get; set; }
    public string SKU { get; set; } = string.Empty;
    public string Nome { get; set; } = string.Empty;
    public string? Descricao { get; set; }
    public Guid CategoriaId { get; set; }
    public string UnidadeMedida { get; set; } = "UN";
    public string? CodigoBarras { get; set; }
    public int EstoqueMinimo { get; set; }
}
